Parsers/ASimUserManagement/Parsers/vimUserManagementLinuxAuthpriv.yaml

```diff
@@ -0,0 +1,543 @@
+Product:
+  Name: Microsoft
+Normalization:
+  Schema: UserManagement
+  Version: '0.1.1'
+References:
+- Title: ASIM User Management Schema
+  Link: https://aka.ms/ASimUserManagementDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Ubuntu remote logging
+  Link: https://manpages.ubuntu.com/manpages/lunar/en/man5/rsyslog.conf.5.html 
+Description: |
+  This ASIM parser supports normalizing Linux authpriv logs delivered using Syslog to the ASIM UserManagement normalized schema.
+ParserName: vimUserManagementLinuxAuthpriv
+EquivalentBuiltInParser: _Im_UserManagement_LinuxAuthpriv
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: actorusername_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: targetusername_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventtype_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser = (
+        starttime:datetime=datetime(null), 
+        endtime:datetime=datetime(null),
+        srcipaddr_has_any_prefix: dynamic=dynamic([]), 
+        targetusername_has_any: dynamic=dynamic([]),
+        actorusername_has_any: dynamic=dynamic([]),
+        eventtype_in: dynamic=dynamic([]),
+        disabled:bool=false
+    ) {
+  let ActionLookup = datatable (Action:string, EventType:string)
+  [
+      "added",  "UserAddedToGroup",
+      "removed","UserRemovedFromGroup"
+  ];
+  let SeverityLookup = datatable (SeverityLevel:string, EventSeverity:string)
+  [
+      "info", "Informational",
+      "warn", "Low",
+      "err",  "Medium",
+      "crit", "High"
+  ];    
+  let ItemParser = (T:(SyslogMessage:string,EventTime:datetime,Facility:string,MG:string)) {
+      T
+      | project-away SyslogMessage,EventTime,Facility,MG
+  };
+  union (
+      Syslog
+      | where not(disabled)
+      | where Computer in (_ASIM_GetSourceBySourceType('LinuxAuthpriv'))
+      | where (isnull(starttime) or TimeGenerated >= starttime)
+           and (isnull(endtime) or TimeGenerated <= endtime)
+      | where Facility == "authpriv"
+      | where (array_length(actorusername_has_any) == 0 and 
+          array_length(targetusername_has_any) == 0 or (SyslogMessage has_any(targetusername_has_any))) and
+          (array_length(srcipaddr_has_any_prefix) == 0) and
+          (array_length(eventtype_in) == 0 or ("UserCreated" in (eventtype_in)))
+      | where ProcessName == "useradd"
+          and  SyslogMessage startswith "new user: name="
+      | parse SyslogMessage with "new user: name=" TargetUsername ", UID=" TargetUserId ", GID=" GroupId ", " *
+      | where (array_length(targetusername_has_any) == 0 or (TargetUsername has_any(targetusername_has_any))) and
+          array_length(actorusername_has_any) == 0 
```

Comment:
can we check for Actor info for user activity events?

Reply:
Only verified Actor info where there is a TargetUsername and an ActorUsername